File: ObjectiveCProject/ObjectiveCProject/ViewController.m
#import "ViewController.h"
#import <ObjectiveCFramework/ObjectiveCFramework.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (IBAction)loadObjectiveCVC:(id)sender {
    [ObjectiveCManager presentObjectiveCViewController:@"Presented ObjCVC from app" presentingController:self completion:^(UIViewController * _Nullable viewController, NSError * _Nullable error) {
        if (error) {
            NSLog(@"%@", error.description);
            return;
        }
        NSLog(@"Presented objCVC successfully.");
    }];
}

- (IBAction)loadSwiftVCTapped:(id)sender {
    [ObjectiveCManager presentSwiftViewController:@"Presented Swift VC from app" presentingController:self completion:^(UIViewController * _Nullable viewController, NSError * _Nullable error) {
        if (error) {
            NSLog(@"%@", error.description);
            return;
        }
        NSLog(@"Presented swiftVC successfully.");
    }];
}

@end

File: SwiftProject/SwiftProject/ViewController.swift
import ObjectiveCFramework
import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        print("%d", ObjectiveCManager.swiftEnum())
    }

    @IBAction func loadObjCVCTapped(_ sender: Any) {
        ObjectiveCManager.presentObjectiveCViewController("Loading ObjC VC from Swift app", presenting: self) { (_, error) in
            if let error = error {
                print(error)
                return
            }
            print ("Presented objCVC successfully.")
        }
    }
    
    @IBAction func loadSwiftVCTapped(_ sender: Any) {
        ObjectiveCManager.presentSwiftViewController("Presented Swift VC from app", presenting: self) { (_, error) in
            if let error = error {
                print(error)
                return
            }
            print ("Presented swiftVC successfully.")
        }
    }
}

